New LDAP config howto (#6438)
Clinical-Genomics/scout: docs/admin-guide/login-system.md

Institutional directory services authentication via LDAP is supported by Scout.

> LDAP authentication in Scout is achieved by using the [flask-ldap3-login](https://flask-ldap3-login.readthedocs.io/en/latest/) library.

Reviewer: 💯

LDAP server instances are different from case to case. Some basic LDAP config options (with example values) that can be used on the in the Scout config file are the following:

### LDAP Authentication Setup (Test Configuration)

> For development and testing purposes, login system can be connected to a public LDAP test server provided by Forum Systems.

Reviewer: ```suggestion
For development and testing purposes, the login system can be connected to a public LDAP test server.
```

> The following free LDAP server can be used for testing: https://www.forumsys.com/2022/05/10/online-ldap-test-server/

Reviewer: ```suggestion
The following free LDAP server provided by Forum Systems can be used for testing: https://www.forumsys.com/2022/05/10/online-ldap-test-server/
```

This server provides a set of predefined users (e.g. einstein, tesla, newton) with a common password.

Configuration:
To use the test LDAP server as a login system, in the scout config file, un-comment the provided test settings

```
# LDAP login Settings
LDAP_HOST = "ldap.forumsys.com"
LDAP_BIND_USER_DN = "cn=read-only-admin,dc=example,dc=com"
LDAP_BIND_USER_PASSWORD = "password"
LDAP_BASE_DN = "dc=example,dc=com"
LDAP_USER_DN = ""
LDAP_USER_RDN_ATTR = "uid"
LDAP_USER_LOGIN_ATTR = "uid"
LDAP_GROUP_DN = ""
LDAP_GROUP_OBJECT_FILTER = "(objectclass=*)"
```

Test users can be the following usernames:
- einstein
- newton
- tesla
- gauss
- euler

All users share the password: `password`

In addition to LDAP authentication, a corresponding user must exist in Scout for the login to fully work.
Create a matching Scout user with the same username using:

```bash
scout --demo load user -i cust000 -u "Isaac Newton" -m newton@email.com -id newton
```

This ensures that the LDAP identity can be mapped correctly to a Scout user record.

> It should be now possible to login the new user into Scout using username `newton` and password `password`

Reviewer: ```suggestion
It should now be possible to log the new user in to Scout using username `newton` and password `password`
```

### Considerations regarding the LDAP login in Scout.

> Since Scout software needs to define internal user objects for functioning properly, the LDAP login check against the server is only the first step of the login process. The second step would be making sure that a user authenticated by a certain password exists also in the `user` collection of the Scout database. Because of this latter authentication step, it is not strictly necessary to define a stringent set of rules (for instance which groups the user must belong to) for the LDAP login system, because a user from any group will be logged into Scout, as long as it's previously saved in the Scout database.

Reviewer: ```suggestion
Since the Scout software needs to define internal user objects, the LDAP login check against the server is only the first, authentication, step of the login process. The second, authorisation, step would be making sure that a user authenticated by a certain password exists also in the `user` collection of the Scout database. Because of this latter authorisation step, it is not strictly necessary to define a stringent set of rules (for instance which groups the user must belong to) for the LDAP login system, because a user from any group will be logged into Scout, as long as it's previously saved in the Scout database.
```

> This entails that if a user should be removed from the system, it is enough just to remove it from the Scout database.

Reviewer: ```suggestion
This means that if a user should be removed from the system, it is enough just to remove it from the Scout database.
```
